src/main/java/jenkins/plugins/git/GitSCMSource.java

```diff
@@ -626,14 +626,17 @@ public boolean isMatch(SCMSource source) {
                                         return Collections.emptyMap();
                                     }
                                     if (GitStatus.looselyMatches(u, remote)) {
-                                        SCMHead head = new SCMHead(branch);
+                                        GitBranchSCMHead head = new GitBranchSCMHead(branch);
```

Comment:
All pre-filters already have to deal with subclassing by GitHub / Gitea / Bitbucket etc so this subclassing should not affect any extension plugins